external-crates/move/crates/move-core-types/src/resolver.rs

```diff
@@ -40,7 +33,14 @@ pub struct SerializedPackage {
     pub linkage_table: BTreeMap<AccountAddress, AccountAddress>,
     /// The type origin table for the package. Every type in the package must have an entry in this
     /// table.
-    pub type_origin_table: Vec<TypeOrigin>,
+    /// This is a mapping of (module name, type name) to the defining ID of the type -- this is the
+    /// version ID of the package where the type was first defined.
+    pub type_origin_table: BTreeMap<IntraPackageName, AccountAddress>,
+    /// The version number of this package. This is a monotonically increasing number that is used
+    /// to track the version of the package. This is not used for linkage or identification, but
+    /// can be used for other purposes such as displaying the version of the package or determining
+    /// relative ordering of versions of the same package.
+    pub version: u64,
```

Comment:
This design makes me nervous. I wish we could derive this version number directly from the bytes of the serialized package, the linkage provided, or something else. As it stands, we are just counting on doing the right thing at every step.

Reply:
Yea, I wish we could take a `MovePackage` at this point and then convert it from there, but sadly no way of doing that because of how deps are all structured :'(